teps/0008-support-knative-service-for-triggers-eventlistener-pod.md

```diff
@@ -0,0 +1,318 @@
+Also Knative handles k8s service creation by default with public accessibility which in turn available as part of eventlistener status address.
+
+##### Note:
+Triggers provide flexibility to the user to deploy Knative Service but not the installation of Knative itself and its responsibility of the user to have Knative running beforehand.
+
+### Usage examples
+
+#### Default EventListener yaml
+For Backward compatibility the default behavior will be as it is for few releases.
+
+```yaml
+
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+ name: github-listener-interceptor
+spec:
+ serviceAccountName: tekton-triggers-example-sa
+ serviceType: NodePort
+ podTemplate:
+   nodeSelector:
+     app: test
+   tolerations:
+   - key: key
+     value: value
+     operator: Equal
+     effect: NoSchedule
+ triggers:
+   - name: foo-trig
+     interceptors:
+       - github:
+           secretRef:
+             secretName: foo
+             secretKey: bar
+           eventTypes:
+             - pull_request
+     bindings:
+       - ref: pipeline-binding
+     template:
+       name: pipeline-template
+``` 
+
+#### Kubernetes Based 
+This is exactly the same whatever we have right now with default.
+The reason to move `serviceAccountName`, `podTemplate`, 
+to `typed` field is because those are part of [WithPod{}](https://github.com/knative/pkg/blob/master/apis/duck/v1/podspec_types.go#L41) duck type
+and which helps to support any of the pod and container field without hardcoding in [podTemplate](https://github.com/tektoncd/triggers/blob/master/pkg/apis/triggers/v1alpha1/event_listener_types.go#L62).
+
+1.If user specify podSpec fields.  
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+ name: github-listener-interceptor
+spec:
+ triggers:
+   - name: foo-trig
+     interceptors:
+       - github:
+           secretRef:
+             secretName: foo
+             secretKey: bar
+           eventTypes:
+             - pull_request
+     bindings:
+       - ref: pipeline-binding
+     template:
+       name: pipeline-template
+ service:
+   typed:
+     serviceType: NodePort
+     spec:
+       template:
+         metadata:
+           annotations:
+             k8s.based.annotation: "value"
+         spec:
+           serviceAccountName: tekton-triggers-github-sa
+           nodeSelector:
+             app: test
+           tolerations:
+           - key: key
+             value: value
+             operator: Equal
+             effect: NoSchedule
+```
+
+2.If user wants go with default values of podSpec fields then no need to specify `service` in that case trigger deploy 
+kubernetes deployment with default values and yaml looks something like below.
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+ name: github-listener-interceptor
+spec:
+ triggers:
```

Comment:
Nice! Though I think `serviceAccountName` might be a field that would be commonly added too:


```
service:
  typed:
    spec:
      template:
        spec:
          serviceAccountName: "sa"
```

Reply:
I gave that example above i mean [here](https://github.com/tektoncd/community/pull/162/files#diff-7380de68d8a45b8b47cfa45d2fde784eR164) 

This example is to show that if user doesn't specify `service` field then they will get default values including SA as `default`